File: dotNet/Tests/ISP/Example1/JsonStoreTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Solid.ISP.Example1;

namespace Tests.ISP.Example1
{
    [TestClass]
    public class JsonStoreTest
    {
        private JsonStore _subject;

        [TestInitialize]
        public void Setup()
        {
            _subject = new JsonStore();
            _subject.Open();
        }

        [TestCleanup]
        public void Teardown()
        {
            _subject.Close();
        }

        [TestMethod]
        public void Test()
        {
            string result = _subject.ReadFile();

            Assert.AreEqual("File content!", result);
        }
    }
}

File: java/isp-start/src/test/java/no/bouvet/solid/isp/JsonStoreTest.java
package no.bouvet.solid.isp;

import static org.junit.Assert.assertEquals;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class JsonStoreTest {

    private JsonStore subject;

    @Before
    public void beforeTest() {
        subject = new JsonStore();
        subject.open();
    }

    @After
    public void afterTest() {
        subject.close();
    }

    @Test
    public void test() {
        // when
        String result = subject.readFile();

        // then
        assertEquals("File content!", result);
    }

}
